src/project-tests/pomodoro-clock-tests.js

```diff
@@ -78,476 +84,718 @@ export default function createPomodoroClockTests() {
-            const breakLength = document.getElementById("break-length");
-            if (timeLeft.innerHTML === "00:00") {
+          // Since not requiring specific labels, save the 'session' label to a 
+          // variable, then test within observer function that label has changed
+          // to know when in break
+          let sessionLabel = document.getElementById('timer-label').innerHTML;
+          const observer = observeElement('time-left', () => {
+            const currentTimer = document.getElementById('timer-label');
+            const breakLength = document.getElementById('break-length');
+            if (timeLeft.innerHTML === '00:00') {
               shouldBeInBreak = true;
-            } else {
-              if (parseInt(timeLeft.innerHTML.slice(0, 2)) > 5) {
-                reject(new Error("Timer has switched to Break time, but it didn't start with the correct value: " + (parseInt(getMinutes(timeLeft.innerHTML)) + 1) + " instead of " + breakLength.innerHTML));
-              } else if (shouldBeInBreak) {
-                if (currentTimer.innerHTML !== sessionLabel) {
-                  if (+ getMinutes(timeLeft.innerHTML) === + breakLength.innerHTML)
-                    resolve();
-                  else {
-                    reject(new Error("Timer has switched to Break time, but it didn't start with the correct value: " + getMinutes(timeLeft.innerHTML) + " instead of " + breakLength.innerHTML));
-                  }
+            } else if (parseInt(timeLeft.innerHTML.slice(0, 2), 10) > 5) {
+              reject(
+                new Error('Timer has switched to Break time, but it didn\'t ' +
+                'start with the correct value: ' + (
+                  parseInt(getMinutes(timeLeft.innerHTML), 10) + 1
+                ) + ' instead of ' + breakLength.innerHTML)
+              );
+              observer.disconnect();
+
+            } else if (shouldBeInBreak) {
+              if (currentTimer.innerHTML !== sessionLabel) {
+                let getTimeLeftHTML = +getMinutes(timeLeft.innerHTML);
+                if (getTimeLeftHTML === +breakLength.innerHTML) {
+                  resolve();
                 } else {
-                  reject(new Error("Timer has reached zero but didn't switch to Break time"));
+                  reject(
+                    new Error('Timer has switched to Break time, but it ' +
+                    'didn\'t start with the correct value: ' +
+                    getMinutes(timeLeft.innerHTML) + ' instead of ' +
+                    breakLength.innerHTML)
+                  );
+                  observer.disconnect();
                 }
-                observer.disconnect();
+              } else {
+                reject(
+                  new Error('Timer has reached zero but didn\'t switch to ' +
+                  'Break time')
+                );
               }
+              observer.disconnect();
```

Comment:
Same thing here. Seems like when observer.disconnect() wasn't called then subsequent tests failed.

Reply:
Ok, please just put a TODO comment in the code to review this later.